### PR TITLE
Pass on ValidationException info to user

### PR DESCRIPTION
**Pull request by dominikl**
Implements [Ticket 13111](https://trac.openmicroscopy.org/ome/ticket/13111)

Test: Try to upload a faulty script. You should get an error dialog providing some information what's wrong with the script.

### CODE REVIEW

**Comment by pwalczysko** [comment]
This works as described in the PR. Nevertheless, I have 2 issues with the workflow:
1. The Upload Script dialog is not good
   - it encourages you to ``Save File`` instead to Upload one
   - it does not show valid scripts in the browsing windows as black, instead you have to click on greyed-out items (see screenshot).

2. The small message in bottom right corner is gone which said: ``Unable to upload script`` 
    - it was telling me that the script was not uploaded
    - now I have the dialog saying ``Script could not be validated.`` - but was it uploaded ? (I know, it was not, but this is not very clear now)

**Comment by dominikl** [comment]
Simplified the appearance of the script dialog a bit. ```ScriptUploaderDialog``` has a lot of unused code. Is it worth cleaning it up @jburel ? Or reenabling it's unused functionality (looks like it was much more powerful at some point in the past)?

**Comment by pwalczysko** [comment]
All works fine. The dialog is logical now. The message says that the script was rejected. 

Ready to merge.

File: components/insight/SRC/org/openmicroscopy/shoola/agents/util/ui/ScriptUploaderDialog.java [hunk 1]
```diff
@@ -285,19 +281,19 @@ private JPanel buildControls()
...
-		JLabel l = UIUtilities.setTextFont("Folder:");
-		details.add(l, "0, "+row+", LEFT, CENTER");
-		details.add(location, "2, "+row);
-		row++;
-		details.add(locationFinder, "0, "+row+", LEFT, CENTER");
-		JXTaskPane pane = new JXTaskPane();
-		pane.setCollapsed(true);
-		pane.setTitle("Script details");
-		pane.add(details);
+//		JLabel l = UIUtilities.setTextFont("Folder:");
```
Comment (joshmoore) [inline components/insight/SRC/org/openmicroscopy/shoola/agents/util/ui/ScriptUploaderDialog.java]: Drop commented code?

**Comment by dominikl** [comment]
Removed all the unused code from the ```ScriptUploaderDialog``` . Sorry @pwalczysko , probably should be tested again.

**Comment by joshmoore** [comment]
@dominikl : leave the scope of this clean up to you since https://github.com/dominikl/openmicroscopy/commit/57dfab28d5c36a070ff407e2840a927633491b33 makes sense, but I would think a commit that just removes a comment & passes travis doesn't need re-listing.

**Comment by pwalczysko** [comment]
Sorry, maybe I missed this previously but there are many more submenu items now as compared with the eel latest (without this PR) after a successful upload of the script. I think this came after the last commit(s) here, did not see it during the first testing here on this PR.

eel merge
screen shot 2015-12-04 at 12 15 28


eel latest
screen shot 2015-12-04 at 12 15 15

**Comment by dominikl** [comment]
You're right, well spotted. I'll fix that.

**Comment by dominikl** [comment]
Sorry @pwalczysko this became a complete overhaul of the ScriptUploaderDialog again, so you'll have to retest again. I added the "Folder" parameter to the ScriptUploaderDialog again (it was there before, but e. g. the "Find folder" button never worked). You can specify in which folder you want to upload the script, if left empty it uses the folder name of your local filesystem where the scripts is (but just the direct parent folder, not the whole folder structure like introduced with the previous commit).
I totally forgot about the help pages. Is this relevant for the help, screenshots, etc. @gusferguson ?

**Comment by pwalczysko** [comment]
Retested. The submenu is now taken just from the nearest parent folder, just like it was before.
The messages are also okay.
Ready to merge.

**Pull request merged**